src/components/locations/components/NewLocation.js

```diff
@@ -1,79 +1,92 @@
-    // const [state, setstate] = useState(initialState);
-    // const dispatch = useDispatch();
+    const dispatch = useDispatch();
     const { loading } = useSelector(state => state.signIn);
     const { getFieldDecorator, resetFields, validateFields } = form;
+    const handleFormData = formValues => {
+        const formData = new FormData();
+        Object.keys(formValues).forEach(key => {
+            formData.append(key, formValues[key]);
+            if (key === 'images') {
+                formValues[key].forEach(element => {
+                    formData.append('images', element.originFileObj);
+                });
+            }
+        });
+        return formData;
+    };
     const handleSubmit = e => {
         e.preventDefault();
-        validateFields((err, values) => {
+        validateFields(async (err, values) => {
             if (!err) {
-                const formValues = {
-                    email: values.email,
-                    password: values.password,
-                };
-                // dispatch(userSignIn(formValues));
-                resetFields();
+                const newFormData = handleFormData(values);
+                dispatch(newLocations(newFormData));
+                setTimeout(() => {
+                    resetFields();
+                }, 3000);
             }
         });
     };
-    // const fileList = [
-    //     {
-    //         uid: '-1',
-    //         name: 'xxx.png',
-    //         status: 'done',
-    //         url: 'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png',
-    //         thumbUrl: 'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png',
-    //     },
-    //     {
-    //         uid: '-2',
-    //         name: 'yyy.png',
-    //         status: 'error',
-    //     },
-    // ];
     const props = {
-        // action: 'https://www.mocky.io/v2/5cc8019d300000980a055e76',
         listType: 'picture',
-        onChange({ file, fileList }) {
-            if (file.status !== 'uploading') {
-                console.log(file, fileList);
-            }
-        },
-        // defaultFileList: [...fileList],
+        name: 'locationImage',
     };
     const renderFormItem = () => FORM_ITEMS.map(item => (
-        <Form.Item key={item.formControlName} hasFeedback>
+        <Form.Item label={item.placeholder} key={item.formControlName} hasFeedback>
             {getFieldDecorator(item.formControlName, {
                 rules: [{ message: item.message, required: item.required }],
-            })(<Input
-                type={item.type}
-                placeholder={item.placeholder}
-            />)}
+            })(item.type === 'textarea' ? (
+                <Input.TextArea />)
+                : (<Input type={item.type} />))}
         </Form.Item>
     ));
+    const dummyRequest = ({ onSuccess }) => {
+        setTimeout(() => {
+            onSuccess('ok');
+        }, 100);
+    };
```

Comment:
this seems very wrong

Reply:
Absoutely not needed. Was testing out something and forgot to remove it at the end.